augly/image/functional.py

```diff
@@ -1759,6 +1759,63 @@ def shuffle_pixels(
     return imutils.ret_and_save_image(aug_image, output_path)
 
 
+def skew(
+    image: Union[str, Image.Image],
+    output_path: Optional[str] = None,
+    level: float = 1.0,
+    max_level: float = 10.0,
+    axis: str = "x-axis",
+    metadata: Optional[List[Dict[str, Any]]] = None,
+) -> Image.Image:
+
+    """
+    Skew image with respect to X-axis or Y-axis
+
+    @param image: the path to an image or a variable of type PIL.Image.Image
+        to be augmented
+
+    @param output_path: the path in which the resulting image will be stored.
+        If None, the resulting PIL Image will still be returned
+
+    @param level: Level of the operation that will be in [0, The maximum value of level]
+
+    @param max_level: Maximum value that the operation can have. This will be
+        scaled to level / PARAMETER_MAX.
+
+    @param axis: Axis along which the image will be skewed; can be 'x-axis' or 'y-axis'
+
+    @param metadata: if set to be a list, metadata about the function execution
+        including its name, the source & dest width, height, etc. will be appended
+        to the inputted list. If set to None, no metadata will be appended or returned
+
+    @returns: the augmented PIL Image
+    """
+
+    image = imutils.validate_and_load_image(image)
+    func_kwargs = imutils.get_func_kwargs(metadata, locals())
+
+    level = float(np.random.uniform(low=0.1, high=level)) * max_level / 10
```

Comment:
Please don't do any random sampling inside augmentation definitions. Just let users do their own random sampling of `level` before applying the augmentations. Otherwise this function's behavior is not reproducible. Please remove this line & the `max_level` param altogether and just specify what range the user should provide a `level` within so that the `skew` behavior is a useful augmentation.